### PR TITLE
Fixes #7347

### LINKED ISSUE #7347: DiscoveryService start and destroy methods should be called during Client start and shutdown when Discovery SPI is enabled.

### PR DESCRIPTION
**Pull request by kobalski**

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java [hunk 1]
```diff
@@ -193,6 +194,10 @@ public HazelcastClientInstanceImpl(ClientConfig config,
             addressProviders.add(new DiscoveryAddressProvider(discoveryService));
         }
 
+        if (Boolean.valueOf(getClientConfig().getProperty(GroupProperty.DISCOVERY_SPI_ENABLED))) {
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java]: @Donnerbart can't we clean this up?

Comment (Donnerbart) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java]: Yes, this should work in this context:
`if (clientProperties.getBoolean(GroupProperty.DISCOVERY_SPI_ENABLED)) {`

Comment (kobalski) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java]: @Donnerbart `DISCOVERY_SPI_ENABLED` was not defined in `ClientProperty`. I added the definition. Can you look at the PR again?  Is this the correct way?
``` 
if (clientProperties.getBoolean(ClientProperty.DISCOVERY_SPI_ENABLED)) {
       discoveryService.start();
}
```

**Comment by bilalyasar** [comment]
:+1:

File: hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java (hunk 1 above)
Comment (Donnerbart) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java]: I think it should be fine, yes.

**Comment by Donnerbart** [comment]
In the test I just see the `hazelcast.discovery.enabled` property being set on the server config. Is the client side code `discoveryService.start();` executed anyway?

**Comment by kobalski** [comment]
@Donnerbart Good catch. Since both client and server use the same `DiscoveryStrategy`, `start` and `destroy` was called by the server not client. I corrected the test. Increased the latch count to 2 and since `hazelcast.discovery.enabled` is set in both server and client, I expect `start` and `destroy` methods to be called twice. I updated test name accordingly.

**Comment by Donnerbart** [comment]
Okay, then it's a :+1: from me.

I would always run the tests with code coverage in IDEA, so you can easily see if your (new) code is executed. It should not be more work than having the `Coverage` plugin installed and push another button to run the test:

run-with-coverage

**Pull request merged**